desktop-src/ADSI/setting-user-passwords-for-ldap-providers.md

```diff
@@ -15,15 +15,16 @@ To a set a user password, use the [**IADsUser.SetPassword**](/windows/desktop/ap
 
 The LDAP provider for Active Directory uses one of three processes to set the password (third-party LDAP directories such as iPlanet do not use this password authentication process). The method may vary according to the network configuration. The set password methods occur in the following order:
 
--   First, the LDAP provider attempts to use LDAP over a 128-bit SSL connection. For LDAP SSL to operate successfully, the LDAP server must have the appropriate server authentication certificate installed and the clients running the ADSI code must trust the authority that issued those certificates. Both the server and the client must support 128-bit encryption.
--   Second, if the SSL connection is unsuccessful, the LDAP provider attempts to use Kerberos. On Windows 2000, Kerberos may not support cross-forest authentication. Later enhancements to Kerberos support cross-forest authentication.
--   Third, if Kerberos is unsuccessful, the LDAP provider attempts a [**NetUserSetInfo**](/windows/desktop/api/lmaccess/nf-lmaccess-netusersetinfo) API call. In previous releases, ADSI called **NetUserSetInfo** in the security context in which the thread was running, and not the security context specified in the call to [**IADsOpenDSObject.OpenDSObject**](/windows/desktop/api/Iads/nf-iads-iadsopendsobject-opendsobject) or [**ADsOpenObject**](/windows/desktop/api/Adshlp/nf-adshlp-adsopenobject). In later releases, this was changed so that the ADSI LDAP provider would impersonate the user specified in the **OpenDSObject** call when it calls **NetUserSetInfo**.
+-   First, the LDAP provider attempts to use Kerberos. 
+-   Second, if Kerberos is unsuccessful, the LDAP provider attempts to establish a Secure LDAP connection. For Secure LDAP to operate successfully, the LDAP server must have the appropriate server authentication certificate installed and the clients running the ADSI code must trust the authority that issued those certificates. Both the server and the client must support at least 128-bit encryption.
+-   Third, if the Secure LDAP connection is unsuccessful, the LDAP provider attempts a [**NetUserSetInfo**](/windows/desktop/api/lmaccess/nf-lmaccess-netusersetinfo) API call. In previous releases, ADSI called **NetUserSetInfo** in the security context in which the thread was running, and not the security context specified in the call to [**IADsOpenDSObject.OpenDSObject**](/windows/desktop/api/Iads/nf-iads-iadsopendsobject-opendsobject) or [**ADsOpenObject**](/windows/desktop/api/Adshlp/nf-adshlp-adsopenobject). In later releases, this was changed so that the ADSI LDAP provider would impersonate the user specified in the **OpenDSObject** call when it calls **NetUserSetInfo**.
```

Comment:
```suggestion
-   Third, if the Secure LDAP connection is unsuccessful, the LDAP provider attempts a [**NetUserSetInfo**](/windows/win32/api/lmaccess/nf-lmaccess-netusersetinfo) API call. In previous releases, ADSI called **NetUserSetInfo** in the security context in which the thread was running, and not the security context specified in the call to [**IADsOpenDSObject.OpenDSObject**](/windows/win32/api/Iads/nf-iads-iadsopendsobject-opendsobject) or [**ADsOpenObject**](/windows/win32/api/Adshlp/nf-adshlp-adsopenobject). In later releases, this was changed so that the ADSI LDAP provider would impersonate the user specified in the **OpenDSObject** call when it calls **NetUserSetInfo**.
```